src/dft/backends/mklgpu/descriptor.cpp

```diff
@@ -28,7 +28,10 @@ namespace dft {
 
 template <precision prec, domain dom>
 void descriptor<prec, dom>::commit(backend_selector<backend::mklgpu> selector) {
-    pimpl_.reset(mklgpu::create_commit(*this, selector.get_queue()));
+    if (!pimpl_) {
+        pimpl_.reset(mklgpu::create_commit(*this, selector.get_queue()));
+    }
+    pimpl_->commit(values_);
```

Comment:
We need to make sure that this isn't called on a descriptor that was previously for a different backend or queue.

Reply:
Same as the generic version.

Reply:
I don't understand your reply. Right now, if I've commited with `backend_selector<mklcpu>` then `pimpl_` will have a value, so only the values will change. Likewise, I can change the queue for `mklgpu` and that won't get passed onto the backend commit impl.

Reply:
In my previous comment, I think I must have been looking at the uncorrected version of the code. Apologies for the confusion.